tests/ui/binding/irrefutable-in-let-chains.rs

```diff
@@ -0,0 +1,134 @@
+//@ edition: 2024
+//@ check-pass
+
+#![feature(if_let_guard)]
+
+use std::ops::Range;
+
+fn main() {
+    let opt = Some(None..Some(1));
+
+    if let first = &opt {}
+    //~^ WARN irrefutable `if let` pattern
+
+    if let first = &opt
+        && let Some(second) = first
+    {}
```

Comment:
Please add a comment at the top of this test describing the purpose of this test overall and comments throughout describing the nature of the cases that we're testing.